### PR TITLE
UI Improvements and Test refactoring, bug fixes

### PR DESCRIPTION
**Pull request by nbuonin**
This PR:
- Conditionally renders UI elements to make the flow of the app easier for users.
- Refactors tests for the above
- Fixes errors caused by out of range indexing
- Adds an error boundary to catch errors in the app, and displays a message to the user.

### CODE REVIEW

**Comment by nikolas** [comment]
It's much more clear to use with these changes. As a user I would expect the Reset button to also reset:

* Seed
* Distribution Type
* Mean
* Standard Deviation

It's going to be less work in the long run, and simpler code, to follow this pattern for your reset button in CentralLimitGraph.jsx: http://www.columbia.edu/~njn2118/journal/2018/8/11.html

So, the code in handleResetSamples() can be changed to `this.setState(this.initialState);` to reset everything.

**Comment by nbuonin** [comment]
That's a great pattern, thanks!

File: src/central_limit_theorem/CentralLimitGraph.jsx [hunk 1]
```diff
@@ -38,23 +38,22 @@ export class CentralLimitGraph extends Component {
         this.generatePopulation = this.generatePopulation.bind(this);
         this.handleGeneratePopulation = this.handleGeneratePopulation
             .bind(this);
+        this.handleSampleForm = this.handleSampleForm.bind(this);
         this.handleResetPopulation = this.handleResetPopulation.bind(this);
```
Comment (nikolas) [inline src/central_limit_theorem/CentralLimitGraph.jsx]: This function can be removed since it's unused now.

Comment (nbuonin) [reply src/central_limit_theorem/CentralLimitGraph.jsx]: Got it. I pulled it.

**Pull request merged**